packages/apollo-cli/src/config.ts

```diff
@@ -0,0 +1,267 @@
+import { basename, dirname, join, relative } from "path";
+import { fs, withGlobalFS } from "apollo-codegen-core/lib/localfs";
+
+import * as fg from "glob";
+import * as minimatch from "minimatch";
+import {
+  GraphQLSchema,
+  buildClientSchema,
+  extendSchema,
+  visit,
+  buildASTSchema
+} from "graphql";
+import { loadSchema } from "./load-schema";
+import { loadQueryDocuments } from "apollo-codegen-core/lib/loading";
+
+export interface EndpointConfig {
+  url?: string; // main HTTP endpoint
+  subscriptions?: string; // WS endpoint for subscriptions
+  headers?: Object; // headers to send when performing operations
+}
+
+export interface SchemaDependency {
+  schema?: string;
+  endpoint?: EndpointConfig;
+  engineKey?: string;
+  extends?: string;
+  clientSide?: boolean;
+}
+
+export interface DocumentSet {
+  schema?: string;
+  includes: string[];
+  excludes: string[];
+}
+
+export interface ApolloConfig {
+  projectFolder: string;
+  projectName?: string;
```

Comment:
can we change this to be just `name`?

What is the goal of `projectFolder`?

Reply:
`projectFolder` tracks where the root of the project is for handling relative glob paths. It can't be picked up from `cwd` always (in VSCode for example), so we treat it as a config value.